### PR TITLE
[2.3.0] tenant_route() helper

### PR DESCRIPTION
**Pull request by stancl**
```php
tenant_route('foo'); // current hostname
tenant_route('bar', ['a' => 'b']); // current hostname
tenant_route('abc', [], 'anotherdomain.com');
```

### CODE REVIEW

File: src/helpers.php [hunk 1]
```diff
@@ -29,24 +29,38 @@ function tenant($key = null)
...
-if (! \function_exists('global_cache')) {
+if (! function_exists('global_cache')) {
     function global_cache()
     {
         return app('globalCache');
     }
 }
+
+if (! function_exists('tenant_route')) {
+    function tenant_route(string $route, array $parameters = [], string $domain = null): string
```
Comment (viezel) [inline src/helpers.php:55]: Strict mode improvement: 

Please specify that `$domain` can be nullable. 

```php
..., ?string $domain = null)
```

Comment (stancl) [reply src/helpers.php:55]: This is not needed. Setting the default value to null automatically makes the type nullable.

**Comment by ahmedsayedabdelsalam** [comment]
I read the docs. if i need to store subdomain i need to store `foo` in domain column in domains table right? not the full domains structure.

but when i use `tenant_route('foo', 'dashboard')` it returns `https://foo/dashboard`  not `https://foo.example.test/dashboard`

**Comment by dylanBcp** [comment]
+1 Same problem as @ahmedsayedabdelsalam

**Comment by dylanBcp** [comment]
If you want to redirect to other subdomain use : 

```php
    $domain = 'foo';
    $domain = Domain::domainFromSubdomain($domain);
    tenant_route($domain, 'dashboard')
```

**Comment by stancl** [comment]
`tenant_route()` expects hostname. It can't build a URL by combining the tenant's `domain` (subdomain) with a central domain, as there may be an unlimited number of central domains.

In setups that have more strictly defined how central domains work, e.g. by limiting them to one domain like our boilerplate, you can use things like `domainFromSubdomain()` (which itself is a boilerplate abstraction, not a Tenancy abstraction) yeah.

**Pull request merged**